x/slashing/keeper/signing_info.go

```diff
@@ -209,3 +243,44 @@ func (k Keeper) GetValidatorMissedBlocks(ctx context.Context, addr sdk.ConsAddre
 
 	return missedBlocks, err
 }
+
+// PerformConsensusPubKeyUpdate updates cons address to its pub key relation
+// Updates signing info, missed blocks (removes old one, and sets new one)
+func (k Keeper) PerformConsensusPubKeyUpdate(ctx context.Context, oldPubKey, newPubKey cryptotypes.PubKey) error {
```

Comment:
private?

Reply:
We are using map look-up only for missed blocks, we need to keep this method for migrating the signing-info (which has one entry for a validator).